components/encryption/src/manager/mod.rs

```diff
@@ -522,12 +608,12 @@ impl EncryptionKeyManager for DataKeyManager {
     }
 
     fn new_file(&self, fname: &str) -> IoResult<FileEncryptionInfo> {
-        let mut dicts = self.dicts.write().unwrap();
         // Rotate data key if necessary.
-        dicts.maybe_rotate_data_key(self.method, self.master_key.as_ref())?;
-        let (_, data_key) = dicts.current_data_key();
+        self.dicts
+            .maybe_rotate_data_key(self.method, self.master_key.as_ref())?;
```

Comment:
remove this. with the background thread we don't need `new_file` to check key rotation, and get some latency when key rotation (extra IO) is needed. it is okay we use a slightly stale key.